fix: Item Tax Template by Company Permissions

ahmadpak opened the pull request:
Currently if multiple companies are being operated and Item Tax Template is created for each one. The user with a specific company permission is still able to see all the Item Tax Templates. This fix ensures that the appropriate Item Tax templates are applied

![image](https://user-images.githubusercontent.com/7881486/85220886-9bfb7300-b3c8-11ea-8ee4-d67cdaede5d9.png)
![image](https://user-images.githubusercontent.com/7881486/85220893-b03f7000-b3c8-11ea-903b-9e71632afe2a.png)
![image](https://user-images.githubusercontent.com/7881486/85220905-c1887c80-b3c8-11ea-821e-403f3ed51aea.png)

vishdha committed:
feat: Taxjar Integration Added

ahmadpak committed:
Module Import Fix

barredterra committed:
fix(report view): explicitly set column width for

barredterra committed:
refactor: query meta data only once

barredterra committed:
fix: truncate account names to max length

barredterra committed:
fix: customer and supplier data

barredterra committed:
fix: hide transaction-specific for master data

Commit:
fix: product_info

deepeshgarg007 committed:
fix: Minor fixes in loan

ahmadpak committed:
fix: Item Tax Template

ahmadpak committed:
fix: Item Tax Template

ahmadpak committed:
fix: Item Tax Template with no validity

ahmadpak committed:
fix: Item Tax Template with no validity

ahmadpak committed:
fix: Item Tax Template

ahmadpak committed:
fix: Item Tax Template

ahmadpak committed:
Company Field Added in Item Tax Template

ahmadpak committed:
Company Field Added in Item Tax Template

ahmadpak committed:
fix: Item Tax Template check on company field

ahmadpak committed:
fix: Item Tax Template check on company field

ahmadpak committed:
Merge branch 'develop' of github.com:ahmadpak/erpnext into develop

ahmadpak committed:
Merge branch 'develop' of github.com:ahmadpak/erpnext into develop

deepeshgarg007 reviewed (changes requested):
@ahmadpak 
- A patch is needed to update company in the Item Tax Template to update existing records
- Also there is no need to add separate column for company in "Item Tax" table since its fetched from Item Tax Template, can be fetched from Item  Tax Template itself in `_get_item_tax_template`

barredterra committed:
Merge branch 'develop' into refactor_datev

Anurag810 committed:
feat: added year filters

Anurag810 committed:
Fix: if there is no component

Anurag810 committed:
fix: test_tax_for_payroll_period

AfshanKhan committed:
fix: removed condition that considered "Standard working hours" while creating "timesheet" as it was setting wrong time #20848

ahmadpak commented:
Ok.

Anurag810 committed:
fix: 'Condition & Formula Help' appears twice

AfshanKhan committed:
fix: removed "standard working hours" also fixed test cases

Anurag810 committed:
Merge branch 'develop' into salary-structure-help-fix

ahmadpak committed:
Company field removed from Item Taxes Table

ahmadpak committed:
Company field removed from Item Taxes Table

ahmadpak committed:
Patch for Updating item Tax Template Company

ahmadpak committed:
Patch for Updating item Tax Template Company

ahmadpak committed:
Patch file updated

ahmadpak committed:
Patch file updated

ahmadpak committed:
Merge branch 'develop' of github.com:frappe/erpnext into develop

ahmadpak commented:
@deepeshgarg007 Please review the changes. 

Regards

ruchamahabal committed:
fix: handle custom statuses for the pause SLA configuration (#22349)

* fix: handle hold time for custom statuses

* fix: set total hold time only if hold configurations are enabled

Co-authored-by: Himanshu <himanshuwarekar@yahoo.com>

Commit:
fix: setup status indicators for Job Offer and Job Applicant

AfshanKhan committed:
Merge branch 'develop' into FR_ISS_204290

sunhoww committed:
fix: UnboundLocalError when setting product_info (#22436)

Co-authored-by: Marica <maricadsouza221197@gmail.com>

deepeshgarg007 committed:
fix: Add view ledger button for cancelled docs (#22432)

* fix: Add view ledger button for cancelled docs

* fix: Add moddified to date in Stock Ledger

nextchamp-saqib committed:
fix: cannot cancel assets with repair pending (#22440)

* fix: cannot cancel assets with repair pending

* fix: message

* Update asset.py

Co-authored-by: Nabin Hait <nabinhait@gmail.com>

marination committed:
fix: Disable Renaming in Serial No

Commit:
Update customs_tariff_number.json

Main information of tariff numbers are the number. A description is more or less double information as this is mostly the number. Thus a description should not be mandetory.

scmmishra committed:
fix: add company filter in email digest

marination committed:
fix: Handling Empty tables in Production Plan

michellealva committed:
fix: grammatical fix on error message

Before:

Fiscal Year: 2020 does not exists

After:

Fiscal Year 2020 Does Not Exist

rohitwaghchaure committed:
Merge pull request #22463 from frappe/michellealva-fiscal_year

fix: grammatical fix on error message

rohitwaghchaure committed:
Merge pull request #22462 from marination/production-plan-ux

fix: Handling Empty tables in Production Plan

rohitwaghchaure committed:
Merge pull request #22458 from scmmishra/fix-email-digest-so-company

fix: add company filter in email digest

rohitwaghchaure committed:
Merge pull request #22400 from AfshanKhan/FR_ISS_204290

fix: removed condition that considered "Standard working hours" while…

anupamvs committed:
enable show_configure_button when shopping cart is enabled

nabinhait committed:
fix: Send email to credit controllers to increase credit limit

anupamvs committed:
Merge branch 'develop' of https://github.com/frappe/erpnext into shopping-cart-settings

Commit:
Update customs_tariff_number.json

marination committed:
fix: Sales Order to MR test failing

rohitwaghchaure committed:
Merge pull request #22470 from marination/production-plan-test

fix: Sales Order to MR test failing

rohitwaghchaure committed:
Update shopping_cart_settings.js

rohitwaghchaure committed:
Merge pull request #22467 from nabinhait/credit_limit_email

fix: Send email to credit controllers to increase credit limit

rohitwaghchaure committed:
Merge pull request #22468 from frappe/shopping-cart-settings

fix: enable show_configure_button when shopping cart is enabled

michellealva committed:
fix: remove redundant label in salary slip

michellealva committed:
fix: apply filters in all transactions

marination committed:
Merge pull request #22477 from michellealva/label-fix2

fix: remove redundant label in salary slip

marination committed:
Merge branch 'develop' into serial-no-rename

michellealva committed:
fix: change button label in issue

Changed the label of the button from "Make" to "Create" so that it is consistent with other DocTypes.

Before:

deepeshgarg007 committed:
Merge pull request #22481 from frappe/michellealva-patch-7

fix: change button label in issue

deepeshgarg007 committed:
fix: Syntax Error

deepeshgarg007 committed:
Merge branch 'develop' into feat_taxjar

deepeshgarg007 committed:
Merge pull request #22483 from deepeshgarg007/loan_security_shorttfall_fixes

fix: Loan security shortfall fixes

rohitwaghchaure committed:
Merge pull request #22461 from britlog/develop

fix: product_info

rohitwaghchaure committed:
fix: indentation

rohitwaghchaure committed:
Merge pull request #22450 from marination/serial-no-rename

fix: Disable Renaming in Serial No

rohitwaghchaure committed:
Merge pull request #22454 from wojosc/patch-10

refactor: update customs_tariff_number.json

rohitwaghchaure committed:
Merge pull request #22478 from michellealva/cost_center_filter

fix: apply filters in all transactions

rohitwaghchaure committed:
Merge branch 'develop' into dev-job-offer-list-view

rohitwaghchaure committed:
Merge pull request #22397 from Anurag810/salary_deductions_report_fixes

fix: Salary deductions report fixes

rohitwaghchaure committed:
Merge pull request #22413 from Anurag810/salary-structure-help-fix

fix: 'Condition & Formula Help' appears twice

deepeshgarg007 committed:
Merge pull request #22445 from Alchez/dev-job-offer-list-view

fix: setup status indicators for Job Offer and Job Applicant (develop)

deepeshgarg007 committed:
Merge pull request #21690 from alyf-de/refactor_datev

fix(regional): DATEV report

rohitwaghchaure committed:
fix: letter head not found in opening invoice creation tool

deepeshgarg007 commented on erpnext/patches.txt:
```diff
@@ -706,3 +706,6 @@ execute:frappe.delete_doc_if_exists("DocType", "Bank Reconciliation")
 erpnext.patches.v13_0.move_doctype_reports_and_notification_from_hr_to_payroll #22-06-2020
 erpnext.patches.v13_0.move_payroll_setting_separately_from_hr_settings #22-06-2020
 erpnext.patches.v13_0.check_is_income_tax_component #22-06-2020
+execute:frappe.reload_doc("HR", "doctype", "Employee Advance")
```
Remove this two unrelated patches, already there above in patches.txt file

deepeshgarg007 commented on erpnext/patches/v12_0/update_item_tax_template_company.py:
```diff
@@ -0,0 +1,11 @@
+import frappe
+
+def execute():
+    item_tax_template_list = frappe.get_list('Item Tax Template')
```
Reload the `Item Tax Template` doc before running the patch, companies are not getting saved properly in all the templates on running the patch first time, I had to rerun it again manually. Rest all seems fine

ahmadpak replied:
Ok thanks.

deepeshgarg007 committed:
fix: patch

rohitwaghchaure committed:
Merge pull request #22488 from rohitwaghchaure/fixed-letter-head-not-found

fix: letter head not found in opening invoice creation tool

deepeshgarg007 committed:
fix: Patch

deepeshgarg007 committed:
Merge pull request #21047 from vishdha/feat_taxjar

feat(ERPNext Integrations): Taxjar Integration Added

Mangesh-Khairnar committed:
fix: handle nonetype issue for packed items (#22493)

Anurag810 committed:
fix: take parent cost center for child if no cost center at child (#22496)

marination committed:
fix: Stock Onboarding typo and reorder (#22499)

marination committed:
fix: Set Root as Parent if no parent in new tree view node (#22497)

Co-authored-by: Nabin Hait <nabinhait@gmail.com>

Anurag810 committed:
fix: job offer validation fix (#22504)

marination committed:
fix: Added Missing Trends Reports to Buying/Selling Desk (#22502)

* fix: Added Missing Trends Reports to Buying/Selling Desk

* fix: Move Reports to Other Reports

Anurag810 committed:
fixes: Payroll module pre relese test fixes (#22500)

* fix: salary payment based on payments

* fix: bank remittance report

* fix: Payroll period onboarding

* fix: provident-fund-deductions report

* fix: Considered unmarked days

* fix: onboarding paYroll

* feat: quick entry in payroll entry

AfshanKhan committed:
feat: Autofill Supplier pop-up when only 1 Supplier in RFQ

rohitwaghchaure committed:
fix: manufacturing dashboard grammar (#22510)

* fix: manufacturing dashboard grammar

* Update manufacturing.json

nabinhait committed:
fix: Dashboard label in Projects and Assets module (#22517)

sahil28297 committed:
chore: set develop version as 13.x.x

michellealva committed:
fix: Refactor dashboard links in leave policy (#22519)

* fix: refactor dashboard links in leave policy

* fx: code fix

* fix: add labels to links

* fix: code change

marination committed:
Merge branch 'develop' into FR_ISS_204863

marination committed:
Merge pull request #22512 from AfshanKhan/FR_ISS_204863

feat: Autofill Supplier pop-up when only 1 Supplier in RFQ

khushalti committed:
fix: no else condition required since we can decalre it by default (#22537)

* fix: removing unnecessary else condition on customer form

* fix: removing unnecessary else condition on customer form

* fix: no else condition required since we can decalre it by default

Co-authored-by: Khushal Trivedi <khushalt5@gmail.com>

AfshanKhan committed:
made "Subscription Section", "Auto Repeat" and Hub Publishing" collapsible by default (#22521)

Co-authored-by: Marica <maricadsouza221197@gmail.com>

deepeshgarg007 committed:
fix: Remove loan security field from loan

deepeshgarg007 committed:
Merge pull request #22564 from deepeshgarg007/loan_doc_cleanup

fix: Remove loan security field from loan

deepeshgarg007 commented:
@ahmadpak any updates?

ahmadpak commented:
@deepeshgarg007 I will push the changes today. Sorry I have been very busy in past few days.

sahil28297 committed:
Merge branch 'develop' into fix_develop_version

sahil28297 committed:
Merge pull request #22527 from sahil28297/fix_develop_version

chore: set develop version as 13.x.x

ahmadpak committed:
Merge branch 'develop' of github.com:ahmadpak/erpnext into develop

ahmadpak committed:
Patch file updated

ahmadpak committed:
Patch file updated

ahmadpak committed:
Merge commit '6fc2f81ee7' into develop
